variables/builtin.go

```diff
@@ -70,6 +71,15 @@ func SetPlatformArgs(s *Scope, platform specs.Platform) {
 	s.AddInactive("TARGETVARIANT", platform.Variant)
 }
 
+// SetUserPlatformArgs sets the user's platform-specific built-in args.
+func SetUserPlatformArgs(s *Scope) {
+	platform := platforms.DefaultSpec()
```

Comment:
This won't work very well as we default to `linux/*` on a Mac for example. We probably need a new function in `platforms` for this.

Reply:
There's the function `llbutil.DefaultPlatform` that looks to default to `linux/*` for Mac. However, it also does the same for Windows, which I don't think is desired in this case.

Would it be worth creating a different variant that does not change the OS for Windows?

Reply:
Yes, different flavor is this would make sense

Reply:
Actually, your original implementation was correct here. Sorry for misleading. I was thinking of our `llbutil` package, but actually you were correctly using the containerd `platforms`, which doesn't override mac nor windows.

Reply:
Ah ok, no worries.